appserver/web/admin/src/main/java/org/glassfish/web/admin/monitor/WebStatsProviderBootstrap.java

```diff
@@ -65,50 +67,37 @@
  */
 @Service(name = "web")
 @Singleton
-public class WebStatsProviderBootstrap implements PostConstruct {
+public class WebStatsProviderBootstrap implements PostConstruct, MonitoringDataSource {
 
     private static final String NODE_SEPARATOR = "/";
 
-    @Inject
-    private Domain domain;
-
     // Map of apps and its StatsProvider list
     private ConcurrentMap<String, ConcurrentMap<String, Queue<Object>>> vsNameToStatsProviderMap =
-            new ConcurrentHashMap<String, ConcurrentMap<String, Queue<Object>>>();
-    private Queue<Object> webContainerStatsProviderQueue = new ConcurrentLinkedQueue<Object>();
-    private AtomicBoolean isWebStatsProvidersRegistered = new AtomicBoolean(false);
+            new ConcurrentHashMap<>();
+    private Queue<Object> webContainerStatsProviderQueue = new ConcurrentLinkedQueue<>();
 
     public WebStatsProviderBootstrap() {
     }
 
+    @Override
     public void postConstruct(){
         //Register the Web stats providers
         registerWebStatsProviders();
     }
 
     private synchronized void registerWebStatsProviders() {
-        if (isWebStatsProvidersRegistered.get()) {
```

Comment:
FYI: I found an inconsistency in behaviour when investigating why undeloy + deploy would cause configuration level not have the expected effect. The `registerWebStatsProviders` would be called in `postConstruct` and when new application was registered since unregistering an application would remove the providers in case this was the last application. This seems inconsistent since on server startup those provider do exist even with no application. Since there might be something depending on this I decided that it was the best fix to never remove them to be consistent over time.